### ISSUE TITLE
Enhance onboarding and profile creation process

### ISSUE THREAD

**Issue by amsos** [Govrn-HQ/foundation#12]
# Enhance onboarding and profile creation process

## Description
We want to improve the data consistency and reliability of the input data of the user profile creation flow.  We believe this can be done in a way that makes the UX smoother for the DAO users as well.

## Why this is important/beneficial for the Govrn Community
Onboarding is a huge problem in the DAO space, this would huge in helping to improve the experience.

## Proposal Link + Additional Resources
The proposed change is the following:
**Current State:**
1. Kevin Malone asks "What's your twitter handle?"
2. User manually writes and inputs their twitter handle.

**Proposed State:**
1. Kevin Malone asks "What's your twitter handle?  Please connect it by posting `this` message to your twitter account and reply to me with the link.
2. We then automatically link the twitter account to the user profile

Other user profile info to do this for:
- [ ] Wallet
- [ ] Discourse
- [ ] Github

**Comment by anaxareian** [comment]
I don't think this is the best UI/UX at all as described in the proposed state.  I believe a better UI/UX, after getting them to supply their Twitter handle or Twitter URL, would be 

1. Pick one of your favorite of our tweets
2. Quote Tweet it telling us what you liked about it, and 
3. tag us in the quoted retweet.  

Communicate that has to be done within a fixed time, your choice upto  7 days.  Give them that much time to do it, come back and run a '/verify-twitter' command to your Discord bot. 
 that searches for an @mention of Govrn, by the user with their Twitter handle.  And, then confirms them, or indicates something went wrong, but their tweet could not be found.  If the tweeted timely, the latter should never happen, or they supplied the wrong Twitter handle.

This removes a "noisy," unnatural tweet from everyone's timeline, makes verification a little simpler and less error prone, and provides information about the user to Govrn, and small, but real, initial value to Govrn from the beginning of an onboarders desire to say they wan to contribute.  No later than the official end of onboarding, they shold also be acknowledged and appreciated for their contribution. 

My two cents!  (@aia)

> **Proposed State:**
Kevin Malone asks "What's your twitter handle? 
Please connect it by posting this message to your twitter account and reply to me with the link.
We then automatically link the twitter account to the user profile

**Comment by anaxareian** [comment]
With respect to other items you are seeking,  You can create a better UI/UX by having:

  1. them authenticate, even if not needed otherwise, to a web page on your website with their wallet
  2. Only allow access to your Discourse via OAuth2 from Discord, and
  3. invite some simple but real contribution on Github.  

You automate storing ETH address in a DB, you don't need to know their Discourse email, and you can likely automate a response from Github to the required action, or if not, have a similar process to Twitter, with a '/verify-github' command to your Discord bot. 

> Other user profile info to do this for:
>
> Wallet
 Discourse
 Github

**Comment by amsos** [comment]
> * Pick one of your favorite of our tweets
> * Quote Tweet it telling us what you liked about it, and
> * tag us in the quoted retweet.

ooo - I do really like this.   Instead of the generic (post this verification tweet), we actually have them go into our tweets or values and have them retweet something they like.
- It is worth admitting that the usual "noisy" verification tweets can actual do a pretty good job of creating "buzz" and marketing for a project.
- Additionally, having someone go and find a tweet they like is additional effort that could lead to drop off in verification

I wonder if there is a way to combine the concepts of RT a Govrn idea you like, still creating some kind of "disruptive" verification tweet to create buzz, without increasing the workload on the community member?

**Comment by amsos** [comment]
@alexkeating tagging you to pull you in as well

**Comment by anaxareian** [comment]
That's simple!  Just ask them to RT a specific twet.  

I am a big fan of filters, so I'd rather lose people who 
find simple thoughtful tasks as "too much."  I don't know that my view is better, but I know  no one does, 
and I am not a fan of making decisions out of some 
type of fear.  There is difference between prudence 
and fear, and making things super easy to keep numbershigher just seems like fear to me. 

aia--
True wisdom is less presuming than folly. 
The wise man doubteth often, and changeth his mind; 
the fool is obstinate, and doubteth not; he knoweth all things,
but his own ignorance.  

 

    On Thursday, February 24, 2022, 11:20:21 AM GMT-5, amsos ***@***.***> wrote:  
 
 


   
   - Pick one of your favorite of our tweets
   - Quote Tweet it telling us what you liked about it, and
   - tag us in the quoted retweet.


ooo - I do really like this. Instead of the generic (post this verification tweet), we actually have them go into our tweets or values and have them retweet something they like.
   
   - It is worth admitting that the usual "noisy" verification tweets can actual do a pretty good job of creating "buzz" and marketing for a project.
   - Additionally, having someone go and find a tweet they like is additional effort that could lead to drop off in verification

I wonder if there is a way to combine the concepts of RT a Govrn idea you like, still creating some kind of "disruptive" verification tweet to create buzz, without increasing the workload on the community member?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
